### PR TITLE
Add beakerx recipe

### PR DESCRIPTION
**Pull request by gnestor**

### CODE REVIEW

File: recipes/beakerx/pre-unlink.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Uninstall BeakerX notebook extension
+"${PREFIX}/bin/jupyter-nbextension" uninstall beakerx --py --sys-prefix > /dev/null 2>&1
+
+# Uninstall BeakerX kernel specs
+# python setup.py kernels --disable > /dev/null 2>&1
+"${PREFIX}/bin/jupyter-kernelspec" remove clojure cpp groovy java scala sql > /dev/null 2>&1
```
Comment (mwcraig) [inline recipes/beakerx/pre-unlink.sh]: Are all of these kernelspecs installed by beakerx?

File: recipes/beakerx/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  summary: 'Beaker Extensions for Jupyter'
+
+  description: |
+    Beaker Extensions for Jupyter
+  doc_url: https://github.com/twosigma/beakerx
+  dev_url: https://github.com/twosigma/beakerx
+
+extra:
+  recipe-maintainers:
+    - scottdraves
```
Comment (mwcraig) [inline recipes/beakerx/meta.yaml]: I would add yourself as a maintainer to 😀

**Comment by mwcraig** [comment]
Never mind, @scottdraves, just saw the 👍

**Comment by scottdraves** [comment]
thx, will get back to you with a new revision soon.

BeakerNotebook.com
ScottDraves.com
twitter.com/Scott_Draves

**Comment by mwcraig** [comment]
Thanks -- feel free to ping me when you update (I'm typically buried in github notifications...)

File: recipes/beakerx/pre-unlink.sh (hunk 1 above)
Comment (gnestor) [inline recipes/beakerx/pre-unlink.sh]: Yes, they are

**Comment by gnestor** [comment]
@scottdraves I pushed some updates. Can you review and ping Matt when it's ready?

**Comment by scottdraves** [comment]
👍 LGTM

**Comment by mwcraig** [comment]
Looks like the failure on Windows is real. Two options...you could skip the windows build for now by adding

```yaml
- skip: true  # [win]
```

to the build section now and then get windows fixed up in the feedstock (that will require a re-render in feedstock, which is less obscure than it maybe sounds), or you can keep iterating on the recipe.

My recommendation would be to fix things up in the feedstock because you will have merge rights there, but  it is your call.

**Comment by gnestor** [comment]
@mwcraig It looks like we have resolved some issues that we were running into involving the post-link scripts failing. However, it looks like the appveyor build is failing with the following:

```
Traceback (most recent call last):
  File "setup.py", line 115, in <module>
    setup(**setup_args)
  File "C:\bld\beakerx_1503269632691\_b_env\lib\distutils\core.py", line 151, in setup
    dist.run_commands()
  File "C:\bld\beakerx_1503269632691\_b_env\lib\distutils\dist.py", line 953, in run_commands
    self.run_command(cmd)
  File "C:\bld\beakerx_1503269632691\_b_env\lib\distutils\dist.py", line 972, in run_command
    cmd_obj.run()
  File "C:\bld\beakerx_1503269632691\_b_env\lib\site-packages\setuptools\command\install.py", line 61, in run
    return orig.install.run(self)
  File "C:\bld\beakerx_1503269632691\_b_env\lib\distutils\command\install.py", line 563, in run
    self.run_command('build')
  File "C:\bld\beakerx_1503269632691\_b_env\lib\distutils\cmd.py", line 326, in run_command
    self.distribution.run_command(command)
  File "C:\bld\beakerx_1503269632691\_b_env\lib\distutils\dist.py", line 972, in run_command
    cmd_obj.run()
  File "C:\bld\beakerx_1503269632691\_b_env\lib\distutils\command\build.py", line 127, in run
    self.run_command(cmd_name)
  File "C:\bld\beakerx_1503269632691\_b_env\lib\distutils\cmd.py", line 326, in run_command
    self.distribution.run_command(command)
  File "C:\bld\beakerx_1503269632691\_b_env\lib\distutils\dist.py", line 972, in run_command
    cmd_obj.run()
  File "C:\bld\beakerx_1503269632691\_b_env\lib\site-packages\setuptools\command\build_py.py", line 53, in run
    self.build_package_data()
  File "C:\bld\beakerx_1503269632691\_b_env\lib\site-packages\setuptools\command\build_py.py", line 118, in build_package_data
    for package, src_dir, build_dir, filenames in self.data_files:
  File "C:\bld\beakerx_1503269632691\_b_env\lib\site-packages\setuptools\command\build_py.py", line 66, in __getattr__
    self.data_files = self._get_data_files()
  File "C:\bld\beakerx_1503269632691\_b_env\lib\site-packages\setuptools\command\build_py.py", line 83, in _get_data_files
    return list(map(self._get_pkg_data_files, self.packages or ()))
  File "C:\bld\beakerx_1503269632691\_b_env\lib\site-packages\setuptools\command\build_py.py", line 95, in _get_pkg_data_files
    for file in self.find_data_files(package, src_dir)
  File "C:\bld\beakerx_1503269632691\_b_env\lib\site-packages\setuptools\command\build_py.py", line 114, in find_data_files
    return self.exclude_data_files(package, src_dir, files)
  File "C:\bld\beakerx_1503269632691\_b_env\lib\site-packages\setuptools\command\build_py.py", line 198, in exclude_data_files
    files = list(files)
  File "C:\bld\beakerx_1503269632691\_b_env\lib\site-packages\setuptools\command\build_py.py", line 234, in <genexpr>
    for pattern in raw_patterns
  File "C:\bld\beakerx_1503269632691\_b_env\lib\distutils\util.py", line 124, in convert_path
    raise ValueError, "path '%s' cannot be absolute" % pathname
ValueError: path '/' cannot be absolute
Traceback (most recent call last):
  File "C:\Miniconda\Scripts\conda-build-all-script.py", line 5, in <module>
    sys.exit(conda_build_all.cli.main())
  File "C:\Miniconda\lib\site-packages\conda_build_all\cli.py", line 93, in main
    b.main()
  File "C:\Miniconda\lib\site-packages\conda_build_all\builder.py", line 280, in main
    built_dist_location = self.build(meta, build_config)
  File "C:\Miniconda\lib\site-packages\conda_build_all\builder.py", line 209, in build
    conda_build.api.build(meta.meta, config=config)
  File "C:\Miniconda\lib\site-packages\conda_build\api.py", line 97, in build
    need_source_download=need_source_download, config=config)
  File "C:\Miniconda\lib\site-packages\conda_build\build.py", line 1524, in build_tree
    config=config)
  File "C:\Miniconda\lib\site-packages\conda_build\build.py", line 1114, in build
    windows.build(m, build_file, config=config)
  File "C:\Miniconda\lib\site-packages\conda_build\windows.py", line 231, in build
    check_call_env(cmd, cwd=src_dir)
  File "C:\Miniconda\lib\site-packages\conda_build\utils.py", line 628, in check_call_env
    return _func_defaulting_env_to_os_environ(subprocess.check_call, *popenargs, **kwargs)
  File "C:\Miniconda\lib\site-packages\conda_build\utils.py", line 624, in _func_defaulting_env_to_os_environ
    return func(_args, **kwargs)
  File "C:\Miniconda\lib\subprocess.py", line 186, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['cmd.exe', '/c', 'bld.bat']' returned non-zero exit status 1
```

Can you help troubleshoot?

**Comment by gnestor** [comment]
@mwcraig Ah, you beat me to it! I think we are fine with iterating on this in the feedstock. @scottdraves What do you say?

**Comment by mwcraig** [comment]
> Can you help troubleshoot?

Notice I carefully avoided speculating about the cause 😉 

Sounds like something in the files that are trying to be bundled since it is complaining about a path. Took a look at setup.py/setup.cfg and didn't see anything immediately off.  setupbase.py may be the ultimate source.

I'll be offline for ~24 hours (eclipse) starting in ~2 hours, so may be delayed response after that

**Comment by scottdraves** [comment]
we will look into it, thx.

**Comment by gnestor** [comment]
@mwcraig Ok, enjoy the eclipse! I'm in Indonesia so it sounds like I'm gong to miss it ☹️

@scottdraves The recipe was passing before I updated it to use beakerx@0.2.4.dev0, so that suggests that an absolute path of sorts was added between 0.2.0.dev0 and 0.2.4.dev0? Here's a compare view: https://github.com/twosigma/beakerx/compare/afe8330bff41834fe818106fad7213b98157692a...master

**Comment by scottdraves** [comment]
@mwcraig Hi Matt, ready to merge afaik.

**Comment by mwcraig** [comment]
Thanks for your work on this @scottdraves and @gnestor, merging!

**Comment by scottdraves** [comment]
🐬

**Pull request merged**